### PR TITLE
eRegs progressive web app

### PR DESCRIPTION
**Pull request by contolini**
This PR sets the foundation for service workers and general [PWA](https://developers.google.com/web/progressive-web-apps/checklist)-ness of the new eRegs. [sw-precache](https://github.com/GoogleChromeLabs/sw-precache) is used to generate a service worker that caches eRegs pages so that subsequent page loads are loaded from memory instead of the network. It should make the browsing of regulations super duper fast.

Service workers need to reside at the website's root (i.e. `https://consumerfinance.gov/service-worker.js`) so I named it `eregs-service-worker.js` to avoid any confusion. We may want to create new service workers for other parts of cf.gov in the future.

It's not picking up 100% of eReg's assets so the sw-precache config will need some further tweaking. You can verify the manifest and service worker are working by viewing the Application section of chrome dev tools:

screen shot 2018-04-24 at 1 59 23 am

screen shot 2018-04-24 at 1 59 38 am

You can also verify that the `prerender` resource hint is successfully downloading the next reg section page in the background by visiting `chrome://net-internals/#prerender`.

screen shot 2018-04-24 at 2 01 16 am

## Additions

- Custom webpack config file for eRegs
- Service worker generating script in the webpack file
- Unofficial-definitely-not-approved-but-still-cool eRegs icons designed by the amazing @niqjohnson 
- A [web app manifest](https://developers.google.com/web/fundamentals/web-app-manifest/) that tells browsers about the website

## Removals

- eRegs placeholder JS

## Testing

1. `./frontend.sh`
1. Load a new Regulations 3k reg section page.
1. Open the Application tab in chrome dev tools and see if the manifest and service worker are present.

## Todos

- Ensure all assets are being cached.
- Need to create a 200 offline page.
- The site's mobile experience needs to be improved in general.

## Screenshots

Android users will see a popup recommending they "install" eRegs by saving it to their home screen (please forgive the temporary awkward mobile menu).

eregs-pwa

### CODE REVIEW

File: cfgov/cfgov/urls.py [hunk 1]
```diff
@@ -418,6 +418,13 @@ def flagged_wagtail_template_view(flag_name, template_name):
     flagged_url('REGULATIONS3K',
                 r'^regulations3k/',
                 include_if_app_enabled('regulations3k', 'regulations3k.urls')),
+    flagged_url('REGULATIONS3K',
+                r'^eregs-service-worker.js',
+                TemplateView.as_view(
+                template_name='regulations3k/eregs-service-worker.js',
+                content_type='application/javascript'),
+                name='eregs-service-worker.js'),
+
```
Comment (contolini) [inline cfgov/cfgov/urls.py]: @willbarton @higs4281 Is there a better way to serve the service worker? It needs to be accessible at `https://consumerfinance.gov/eregs-service-worker.js`.

File: cfgov/unprocessed/apps/regulations3k/webpack-config.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  return fancyLog( `Service worker file successfully generated to ${ SERVICE_WORKER_DEST }` );
+} );
+
+fancyLog( 'Copying eRegs\' manifest...' );
+fs.copyFile( MANIFEST_SRC, MANIFEST_DEST, err => {
+  if ( err ) {
+    return fancyLog( `Error copying eRegs' manifest: ${ err }` );
+  }
+  return fancyLog( `Successfully copied eRegs' manifest to ${ SERVICE_WORKER_DEST }` );
+} );
```
Comment (contolini) [inline cfgov/unprocessed/apps/regulations3k/webpack-config.js]: @anselmbradford Here's how I ended up getting the files where we need them. Lmk if you see any issues.

File: cfgov/unprocessed/apps/regulations3k/webpack-config.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  return fancyLog( `Service worker file successfully generated to ${ SERVICE_WORKER_DEST }` );
+} );
+
+fancyLog( 'Copying eRegs\' manifest...' );
+fs.copyFile( MANIFEST_SRC, MANIFEST_DEST, err => {
+  if ( err ) {
+    return fancyLog( `Error copying eRegs' manifest: ${ err }` );
+  }
+  return fancyLog( `Successfully copied eRegs' manifest to ${ MANIFEST_DEST }` );
+} );
```
Comment (contolini) [inline cfgov/unprocessed/apps/regulations3k/webpack-config.js]: @anselmbradford Here's how I ended up getting the files where we need them. Lmk if you see any issues.

Comment (anselmbradford) [reply cfgov/unprocessed/apps/regulations3k/webpack-config.js]: Looks good. You may need `// eslint-disable-next-line no-sync` above `fs.copyFile`.

**Comment by anselmbradford** [comment]
Rebase and squash!

**Comment by willbarton** [comment]
> Rebase and squash!

I have my git surgical mask on!

**Comment by willbarton** [comment]
@contolini FYI, I have squashed regulations3k-add-models into a single commit for maximum rebasefulness, and rebased on that.

File: cfgov/unprocessed/apps/regulations3k/webpack-config.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+fancyLog( 'Started generating service worker file...' );
+swPrecache.write( SERVICE_WORKER_DEST, SERVICE_WORKER_CONFIG, err => {
+  if ( err ) {
+    return fancyLog( `Error generating service worker file: ${ err }` );
+  }
+  return fancyLog( `Service worker file successfully generated to ${ SERVICE_WORKER_DEST }` );
+} );
+
+fancyLog( 'Copying eRegs\' manifest...' );
```
Comment (jimmynotjim) [inline cfgov/unprocessed/apps/regulations3k/webpack-config.js]: Might as well use back ticks here since you are elsewhere and avoid escaping the `'`

**Comment by contolini** [comment]
@jimmynotjim We had some rebasing troubles. Those files shouldn't have been deleted. I'm going to close this PR and open a fresh one against master after #4060 is merged.

**Pull request closed without merging**